ddpui/utils/helpers.py

```diff
@@ -182,3 +184,28 @@ def convert_sqlalchemy_rows_to_csv_string(rows: list[dict]):
 def convert_sqlalchemy_rows_to_json_string(rows: list[dict]):
     """converts a list of sqlalchemy rows to a csv string"""
     return json.dumps(convert_to_standard_types(rows), indent=4)
+
+
+def get_schedule_time_for_large_jobs(
+    now: datetime = datetime.now(pytz.utc),
+    time_of_day: time = time(10, 0),
+    weekday: int = calendar.SUNDAY,
+) -> datetime:
+    """By default scheduled time will be Sunday 10am UTC"""
+    # Calculate the number of days until weekday
+    days_until_sunday = (weekday - now.weekday() + 7) % 7
+
+    # If today is that weekday (param), we want to use some future time
+    extra_delta = timedelta(hours=0)
+    if days_until_sunday == 0:
+        extra_delta = timedelta(hours=1)
+        time_of_day = None  # since it will time at now + extra delta
+
+    # Calculate the next weekday datetime
+    next_slot = now + timedelta(days=days_until_sunday) + extra_delta
+
+    # set the time
+    if time_of_day:
+        next_slot = datetime.combine(next_slot, time_of_day, tzinfo=pytz.utc)
+
```

Comment:
also... we should make sure `next_slot` is not in the past, and if it is we should push it by another week

Reply:
its not possible that next_slot is in past.